### PR TITLE
[TESTS #4016] testrail report for extended automated tests

### PR DESCRIPTION
**Pull request by antdanchenko**
added an ability to report automated test results for nightly builds in TestRail

### CODE REVIEW

**Comment by antdanchenko** [comment]
do not merge before #3998

File: test/appium/tests/marks.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import pytest
+
+pr = pytest.mark.pr
+testrail_case_id = pytest.mark.testrail_case_id
+chat = pytest.mark.chat
+all = pytest.mark.all
+transaction = pytest.mark.trnasaction
```
Comment (yevh-berdnyk) [inline test/appium/tests/marks.py]: typo in `trnasaction`

File: test/appium/support/testrail_report.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        self.outcomes = {
+            'passed': 1,
+            'undefined_fail': 10}
+
+        self.headers = dict()
+        self.headers['Authorization'] = 'Basic %s' % str(
+            base64.b64encode(bytes('%s:%s' % (self.user, self.password), 'utf-8')), 'ascii').strip()
+        self.headers['Content-Type'] = 'application/json'
+
+        self.url = 'https://ethstatus.testrail.net/index.php?/api/v2/'
```
Comment (lukaszfryc) [inline test/appium/support/testrail_report.py:29]: Those all (`run_id`, `suite_id`, `project_id`, `url`, ..) should probably be constants. Like `RUN_ID`

**Comment by lukaszfryc** [comment]
@antdanchenko could you paste here a link to some results in TestRail?

File: test/appium/support/testrail_report.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    def get_tests(self, run_id):
+        return self.get('get_tests/%s' % run_id)
+
+    def get_milestones(self, project_id):
+        return self.get('get_milestones/%s' % project_id)
+
+    def add_run(self, run_name):
+        request_body = {'suite_id': self.suite_id,
+                        'name': run_name,
+                        'milestone_id': self.get_milestones(self.project_id)[-1]['id']}
```
Comment (lukaszfryc) [inline test/appium/support/testrail_report.py]: Maybe change `self.get_milestones(self.project_id)[-1]['id']` to `get_last_mailstone_id`

File: test/appium/tests/conftest.py [hunk 4]
```diff
@@ -70,31 +76,34 @@ def pytest_configure(config):
...
-        pull = repo.get_pull(int(config.getoption('pr_number')))
-        pull.create_issue_comment(github_report.build_html_report())
+    if is_master(config):
+        if config.getoption('pr_number'):
+            from github import Github
+            repo = Github(github_token).get_user('status-im').get_repo('status-react')
+            pull = repo.get_pull(int(config.getoption('pr_number')))
+            pull.create_issue_comment(github_report.build_html_report())
+        if config.getoption('nightly'):
+            testrail_report.add_results()
```
Comment (lukaszfryc) [inline test/appium/tests/conftest.py:91]: nice! it's really clean :)

File: test/appium/tests/marks.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import pytest
+
+pr = pytest.mark.pr
+testrail_case_id = pytest.mark.testrail_case_id
```
Comment (lukaszfryc) [inline test/appium/tests/marks.py:4]: I like those aliases!

**Review by lukaszfryc — DISMISSED** [review]
Overall looks great! I added few comments

**Review by yevh-berdnyk — APPROVED** [review]

**Pull request merged**